packages/react/src/table/TableScroller.tsx

```diff
@@ -0,0 +1,333 @@
+			calculateThumbWidth();
+			repositionThumb();
+		});
+
+		observer.observe(scrollerRef.current);
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [calculateThumbWidth, repositionThumb, scrollerRef]);
+
+	const handleThumbPress = (event: React.MouseEvent | React.TouchEvent) => {
+		event.preventDefault();
+
+		if (event.type === 'mousedown' && 'button' in event && event.button === 0) {
+			setIsDraggingThumb(true);
+			mousePos.current = { x: event.pageX, y: event.pageY };
+		} else if (event.type === 'touchstart' && 'touches' in event) {
+			setIsDraggingThumb(true);
+			mousePos.current = {
+				x: event.touches[0].pageX,
+				y: event.touches[0].pageY,
+			};
+		}
+	};
+
+	const handleThumbMove = useCallback(
+		(event: Event | React.MouseEvent | React.TouchEvent) => {
+			if (!scrollerRef?.current) {
+				return;
+			}
+
+			const _pageX = 'pageX' in event ? event.pageX : null;
+			const _pageY = 'pageY' in event ? event.pageY : null;
+			const touches = 'touches' in event ? event.touches : null;
+
+			if (isDraggingThumb) {
+				const pageX = touches ? touches[0].pageX : _pageX;
+				const pageY = touches ? touches[0].pageY : _pageY;
+
+				if (pageX === null || pageY === null) return;
+
+				// Calculate the movement direction
+				const deltaX = pageX - mousePos.current.x;
+				const deltaY = pageY - mousePos.current.y;
+
+				if (Math.abs(deltaX) > Math.abs(deltaY)) {
+					// If horizontal movement is greater than vertical, prevent vertical scrolling
+					event.preventDefault();
+
+					scrollerRef.current.scrollLeft =
+						scrollerRef.current.scrollLeft + deltaX / thumbWidthRatio;
+					mousePos.current.x = pageX;
+				}
+
+				// Update the current Y position to prevent unwanted vertical scrolling
+				mousePos.current.y = pageY;
+			}
+		},
+		[isDraggingThumb, thumbWidthRatio]
+	);
+
+	const handleThumbRelease = useCallback(() => {
+		if (isDraggingThumb) {
+			setIsDraggingThumb(false);
+		}
+	}, [isDraggingThumb]);
+
+	useEffect(() => {
+		if (isDraggingThumb) {
+			document.addEventListener('mousemove', handleThumbMove);
+			document.addEventListener('mouseup', handleThumbRelease);
+			document.addEventListener('touchmove', handleThumbMove);
+			document.addEventListener('touchend', handleThumbRelease);
+		} else {
+			document.removeEventListener('mousemove', handleThumbMove);
+			document.removeEventListener('mouseup', handleThumbRelease);
+			document.removeEventListener('touchmove', handleThumbMove);
+			document.removeEventListener('touchend', handleThumbRelease);
+		}
+
+		return () => {
+			document.removeEventListener('mousemove', handleThumbMove);
+			document.removeEventListener('mouseup', handleThumbRelease);
+			document.removeEventListener('touchmove', handleThumbMove);
+			document.removeEventListener('touchend', handleThumbRelease);
+		};
+	}, [handleThumbMove, handleThumbRelease, isDraggingThumb]);
+
+	const handleScroll = () => {
+		repositionThumb();
+	};
```

Comment:
What do you think about removing the wrapper function `handleScroll` and on line 222 just have:

```ts
onScroll={repositionThumb}
```

Reply:
Much better. d'oh this was a hangover when `handleScroll` was much more complicated